### PR TITLE
:art: Refactor client and server functions

### PR DESCRIPTION
**Pull request by solareon**
## Description

GetPlayers() is no longer an export in the core. The existing function called overlaps with the native. I renamed the function to FindPlayers() and changed the export to the native.

## Checklist

- [X] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [X] My code fits the style guidelines.
- [X] My PR fits the contribution guidelines.

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -185,11 +185,11 @@ end
 
 ---@param src number
 ---@return table
-function GetPlayers(src)
+function FindPlayers(src)
 	local players = {}
 	local playerPed = GetPlayerPed(src)
 	local pCoords = GetEntityCoords(playerPed)
-	for _, v in pairs(exports.qbx_core:GetPlayers()) do
+	for _, v in pairs(GetPlayers()) do
```
Comment (Manason) [inline server/main.lua]: Use https://qbox-project.github.io/resources/core/functions/client#getplayersfromcoords

**Review by Manason — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 1 above)
Comment (solareon) [reply server/main.lua]: Got it. Will refactor it today.

**Comment by solareon** [comment]
Reworked the hire menu logic to use GetPlayersFromCoords. Took me a minute to figure out you need to use GetPlayerServerId() to return the right id. Also fixed some issues with the offline firing and renamed variables to proper case for locals

File: client/cl_utils.lua [hunk 2]
```diff
@@ -7,3 +7,14 @@ function comma_value(amount)
 
     return amount
 end
+
+---@return table
+function FindPlayers()
+	local playerCoords = GetEntityCoords(cache.ped)
+    local closePlayers = {}
+    for _, v in pairs(GetPlayersFromCoords(playerCoords, 10)) do
+        closePlayers[#closePlayers+1] = GetPlayerServerId(v)
```
Comment (Manason) [inline client/cl_utils.lua]: v should already be the serverId, is it not?

File: client/cl_utils.lua [hunk 3]
```diff
@@ -7,3 +7,14 @@ function comma_value(amount)
 
     return amount
 end
+
+---@return table
+function FindPlayers()
+	local playerCoords = GetEntityCoords(cache.ped)
+    local closePlayers = {}
+    for _, v in pairs(GetPlayersFromCoords(playerCoords, 10)) do
```
Comment (Manason) [inline client/cl_utils.lua]: On second thought, if you use https://overextended.dev/ox_lib/Modules/GetNearbyPlayers/Client

Then you can filter out the calling player, since you don't want it in the first place.

File: client/cl_utils.lua [hunk 4]
```diff
@@ -7,3 +7,14 @@ function comma_value(amount)
...
+        closePlayers[#closePlayers+1] = GetPlayerServerId(v)
+    end
+	return lib.callback.await('qb-bossmenu:getplayers', false, closePlayers)
```
Comment (Manason) [inline client/cl_utils.lua]:
```suggestion
	return lib.callback.await('qbx_management:server:getPlayers', false, closePlayers)
```
Would need to be changed elsewhere

File: server/main.lua [hunk 5]
```diff
@@ -183,35 +183,25 @@ function UpdateGrade(src, data, type)
...
-				id = v,
-				coords = GetEntityCoords(targetped),
-				name = ped.PlayerData.charinfo.firstname .. " " .. ped.PlayerData.charinfo.lastname,
-				citizenid = ped.PlayerData.citizenid,
-				sources = GetPlayerPed(ped.PlayerData.source),
-				sourceplayer = ped.PlayerData.source
-			}
-		end
+	for _, v in pairs(closePlayers) do
+		local ped = exports.qbx_core:GetPlayer(v)
```
Comment (Manason) [inline server/main.lua]:
```suggestion
		local player = exports.qbx_core:GetPlayer(v)
```
A ped is a human/animal GTA object. What you are getting here is the qbx_core player object. They aren't the same thing.

File: server/main.lua [hunk 6]
```diff
@@ -183,35 +183,25 @@ function UpdateGrade(src, data, type)
...
+		players[#players + 1] = {
+			id = v,
+			name = ped.PlayerData.charinfo.firstname .. " " .. ped.PlayerData.charinfo.lastname,
+			citizenid = ped.PlayerData.citizenid,
+			sourceplayer = ped.PlayerData.source
```
Comment (Manason) [inline server/main.lua]:
```suggestion
			source = ped.PlayerData.source
```

File: server/sv_boss.lua [hunk 7]
```diff
@@ -50,42 +50,43 @@ end)
...
+		if offlineEmployee[1] then
+			employee = offlineEmployee[1]
+			employee.job = json.decode(employee.job)
+			employee.charinfo = json.decode(employee.charinfo)
+			if employee.job.grade.level > player.PlayerData.job.grade.level then exports.qbx_core:Notify(src, "You cannot fire this citizen!", "error") return end
 			local job = {}
 			job.name = "unemployed"
-			job.label = "Unemployed"
-			job.payment = jobs[job.name].grades['0'].payment or 500
+			job.label = "Civilian"
```
Comment (Manason) [inline server/sv_boss.lua]: Get the label the same way you got the payment; don't hardcode.

**Review by Manason — CHANGES_REQUESTED** [review]

File: client/cl_utils.lua (hunk 2 above)
Comment (solareon) [reply client/cl_utils.lua]: No when I was testing I would get 127,128,129 etc. when my server id was 1,2,3 etc.

Comment (Manason) [reply client/cl_utils.lua]: Disregard. I see you tested it.

File: client/cl_utils.lua (hunk 3 above)
Comment (solareon) [reply client/cl_utils.lua]: I'll give this a go and see if it eliminates the above call to GetPlayerServerId

Comment (Manason) [reply client/cl_utils.lua]: You'll still need to use GetPlayerServerId, but you can get rid of your filtering if condition to filter out the player.

**Comment by solareon** [comment]
Refactored basically the whole thing. All functions and menus were deduplicated and merged into cl_main. Added parameter documentation for functions (I might have gotten this wrong so please let me know). Fixed typos and variable names.

File: server/sv_boss.lua (hunk 7 above)
Comment (solareon) [reply server/sv_boss.lua]: Fixed

File: server/main.lua (hunk 6 above)
Comment (solareon) [reply server/main.lua]: Fixed

File: server/main.lua (hunk 5 above)
Comment (solareon) [reply server/main.lua]: Fixed

File: client/cl_utils.lua (hunk 4 above)
Comment (solareon) [reply client/cl_utils.lua]: Fixed

File: client/cl_utils.lua (hunk 3 above)
Comment (solareon) [reply client/cl_utils.lua]: This did the trick. Still needed the conversion to Server Ids but I got a tighter loop for it.

File: client/cl_main.lua [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+
+---@return table
+function FindPlayers()
+	local playerCoords = GetEntityCoords(cache.ped)
+    local closePlayers = lib.getNearbyPlayers(playerCoords, 10, false)
+    for _, v in pairs(closePlayers) do
+        v.id = GetPlayerServerId(v.id)
+    end
+	return lib.callback.await('qbx_management:server:getplayers', false, closePlayers)
+end
```
Comment (FjamZoo) [inline client/cl_main.lua]:
```suggestion
local function findPlayers()
    local closePlayers = lib.getNearbyPlayers(GetEntityCoords(cache.ped), 10)
    for _, v in pairs(closePlayers) do
        v.id = GetPlayerServerId(v.id)
    end
    return lib.callback.await('qbx_management:server:getplayers', false, closePlayers)
end
```

I would refactor this function slightly, as it's only used in this file use local variable, if something is not used more than a single time like playerCoords was just put it inside whereever it's needed.

File: client/cl_main.lua [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+        options = EmployeeMenu
+    })
+
+    lib.showContext('qbx_management_open_Member')
+end)
+
+---@param type 'job'|'gang'
+RegisterNetEvent('qbx_management:client:HireMenu', function(type)
+    local HireMenu = {}
+    local players = FindPlayers()
```
Comment (FjamZoo) [inline client/cl_main.lua]:
```suggestion
    local players = findPlayers()
```

This would have to be renamed if doing my suggestion above.

File: client/cl_main.lua [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+        id = 'qbx_management_open_Member',
+        title = data.type == 'gang' and 'Manage ' .. data.player.name .. ' - ' .. string.upper(QBX.PlayerData.gang.label) or 'Manage ' .. data.player.name .. ' - ' .. string.upper(QBX.PlayerData.job.label),
+        options = EmployeeMenu
+    })
+
+    lib.showContext('qbx_management_open_Member')
+end)
+
+---@param type 'job'|'gang'
+RegisterNetEvent('qbx_management:client:HireMenu', function(type)
```
Comment (FjamZoo) [inline client/cl_main.lua]: Here I would rename type to, "group" or something that makes it easy for people to see what "type" is as that's very vague, I know the param is there and that's good but we need the variables to have a proper name scheme.

File: server/main.lua [hunk 11]
```diff
@@ -13,136 +14,107 @@ function ExploitBan(id, reason)
...
 	return exports['Renewed-Banking']:getAccountMoney(account) or 0
-end
-
-exports('GetAccount', GetAccount)
-exports('GetGangAccount', GetAccount)
+end)
 
 ---@param account string
 ---@param amount number
 function AddMoney(account, amount)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
local function AddMoney(account, amount)
```

Since we are going to 1 resource file we need all of these functions to be localized

File: server/main.lua [hunk 12]
```diff
@@ -13,136 +14,107 @@ function ExploitBan(id, reason)
...
-
-	Accounts[account] += amount
-	exports['Renewed-Banking']:addAccountMoney(account, amount)
+	exports['Renewed-Banking']:addAccountMoney(account, amount, 'qbx_management')
 end
 
 ---@param account string
 ---@param amount number
 ---@return boolean
 function RemoveMoney(account, amount)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
local function RemoveMoney(account, amount)
```

Since we are going to 1 resource file we need all of these functions to be localized

File: server/main.lua [hunk 13]
```diff
@@ -13,136 +14,107 @@ function ExploitBan(id, reason)
...
 ---@param amount number
----@param pDataType 'job'|'gang'
----@param type 'boss'|'gang'
+---@param type 'job'|'gang'
 ---@param reason string
 ---@return boolean success
 ---@return Player? player populated if successful
 ---@return string? accountName populated if successful
-function WithdrawMoney(src, amount, pDataType, type, reason)
+function WithdrawMoney(src, amount, type, reason)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
local function WithdrawMoney(src, amount, type, reason)
```

Since we are going to 1 resource file we need all of these functions to be localized

Comment (FjamZoo) [inline server/main.lua]: like on the client we should rename type here to group, that way its more specific at a glance and we will have similar naming scheme to client and server.

File: server/main.lua [hunk 14]
```diff
@@ -13,136 +14,107 @@ function ExploitBan(id, reason)
...
 ---@param src number
 ---@param amount number
----@param pDataType 'job'|'gang'
----@param type 'boss'|'gang'
+---@param type 'job'|'gang'
 ---@return boolean success
 ---@return Player? player populated if successful
 ---@return string? accountName populated if successful
-function DepositMoney(src, amount, pDataType, type)
+function DepositMoney(src, amount, type)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
local function DepositMoney(src, amount, type)
```

Since we are going to 1 resource file we need all of these functions to be localized

File: server/main.lua [hunk 15]
```diff
@@ -154,75 +126,219 @@ function GetEmployees(src, accountName, type)
...
-end
+end)
+
+---@param accountName string
+---@param type 'job'|'gang'
+---@return table?
+lib.callback.register('qbx_management:server:getemployees', function(_, accountName, type)
+	local src = source
+	return GetEmployees(src, accountName, type)
+end)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
lib.callback.register('qbx_management:server:getemployees', function(source, accountName, type)
	return GetEmployees(source, accountName, type)
end)
```

When doing callbacks on the server we will always have the source as an arg, which means we do not need to localize it.

File: server/main.lua [hunk 16]
```diff
@@ -154,75 +126,219 @@ function GetEmployees(src, accountName, type)
...
+
+---@param data table
+---@return table?
+RegisterNetEvent('qbx_management:server:GradeUpdate', function(data)
+	local src = source
+	UpdateGrade(src, data)
+	TriggerClientEvent('qbx_management:client:OpenMenu', src, data.type)
+end)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
RegisterNetEvent('qbx_management:server:GradeUpdate', function(data)
	UpdateGrade(source, data)
	TriggerClientEvent('qbx_management:client:OpenMenu', source, data.type)
end)
```

We only want and need to localize source if the coroutine was yielded.

Which in layman terms mean if we use a Wait anywhere

File: server/main.lua [hunk 17]
```diff
@@ -154,75 +126,219 @@ function GetEmployees(src, accountName, type)
...
+---@param amount number
+RegisterNetEvent('qbx_management:server:withdrawMoney', function(type, amount)
+	local src = source
+	local success, player, account = WithdrawMoney(src, amount, type, type == 'gang' and 'Gang menu withdraw' or 'Boss menu withdraw')
+	if not success or not player then return end
+	local logArea = type == 'gang' and 'gang' or 'boss'
+	
+	TriggerEvent('qb-log:server:CreateLog', logArea..'menu', 'Withdraw Money', 'yellow', player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname .. ' successfully withdrew $' .. amount .. ' (' .. account .. ')', false)
+	TriggerClientEvent('qbx_management:client:OpenMenu', src, type)
+end)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
---@param group 'job'|'gang'
---@param amount number
RegisterNetEvent('qbx_management:server:withdrawMoney', function(group, amount)
	local success, player, account = WithdrawMoney(source, amount, group, group == 'gang' and 'Gang menu withdraw' or 'Boss menu withdraw')
	if not success or not player then return end
	local logArea = group == 'gang' and 'gang' or 'boss'
	
	TriggerEvent('qb-log:server:CreateLog', logArea..'menu', 'Withdraw Money', 'yellow', player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname .. ' successfully withdrew $' .. amount .. ' (' .. account .. ')', false)
	TriggerClientEvent('qbx_management:client:OpenMenu', source, type)
end)
```

here we want to rename type to group, to keep going with the naming scheme above.

I also removed the local src = source here because we do not have a yield in any of these functions.

File: server/main.lua [hunk 18]
```diff
@@ -154,75 +126,219 @@ function GetEmployees(src, accountName, type)
...
+---@param amount number
+RegisterNetEvent('qbx_management:server:depositMoney', function(type, amount)
+	local src = source
+	local success, player, account = DepositMoney(src, amount, type)
+	if not success or not player then return end
+
+	local logArea = type == 'gang' and 'gang' or 'boss'
+	TriggerEvent('qb-log:server:CreateLog', logArea..'menu', 'Deposit Money', 'yellow', player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname .. ' successfully deposited $' .. amount .. ' (' .. account .. ')', false)
+	TriggerClientEvent('qbx_management:client:OpenMenu', src, type)
+end)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
---@param group 'job'|'gang'
---@param amount number
RegisterNetEvent('qbx_management:server:depositMoney', function(group, amount)
	local success, player, account = DepositMoney(source, amount, group)
	if not success or not player then return end

	local logArea = group == 'gang' and 'gang' or 'boss'
	TriggerEvent('qb-log:server:CreateLog', logArea..'menu', 'Deposit Money', 'yellow', player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname .. ' successfully deposited $' .. amount .. ' (' .. account .. ')', false)
	TriggerClientEvent('qbx_management:client:OpenMenu', source, group)
end)
```

here we want to rename type to group, to keep going with the naming scheme above.

I also removed the local src = source here because we do not have a yield in any of these functions.

File: client/cl_main.lua [hunk 19]
```diff
@@ -0,0 +1,322 @@
...
+
+---@param type 'job'|'gang'
+RegisterNetEvent('qbx_management:client:SocietyMenu', function(type)
+    local amount = lib.callback.await('qbx_management:server:getAccount', false, QBX.PlayerData[type].name)
+    local SocietyMenu = {
+        {
+            title = 'Deposit',
+            icon = 'fa-solid fa-money-bill-transfer',
+            description = 'Deposit Money',
+            event = 'qbx_management:client:SocietyDeposit',
```
Comment (FjamZoo) [inline client/cl_main.lua]: Here and downwords there's a bunch of "type" that we need to have renamed to group as you have seen in the other messages.

**Review by FjamZoo — CHANGES_REQUESTED** [review]
There's a lot of changes requested here, I could of pointed out everywhere that has "type" that we need to rename to something more appropriate like "group" but I think you get the idea from all the suggestions I've made.

File: server/main.lua (hunk 13 above)
Comment (FjamZoo) [reply server/main.lua]: I forgot to add the suggestion here but if you make global to local functions make sure to camalCase them

Comment (FjamZoo) [reply server/main.lua]: I forgot to add the suggestion here but if you make global to local functions make sure to camalCase them

File: server/main.lua (hunk 14 above)
Comment (FjamZoo) [reply server/main.lua]: I forgot to add the suggestion here but if you make global to local functions make sure to camalCase them

**Comment by Manason** [comment]
I'll add that in the future, please keep your PRs small. This PR is difficult to review now because of it's size.

File: client/cl_main.lua (hunk 8 above)
Comment (solareon) [reply client/cl_main.lua]: Did this one mostly for readability but it's still pretty clear what is going on. Changed

File: client/cl_main.lua (hunk 9 above)
Comment (solareon) [reply client/cl_main.lua]: Got it. Changed.

File: client/cl_main.lua (hunk 10 above)
Comment (solareon) [reply client/cl_main.lua]: Changed. Do agree it makes more sense for group vice type.

File: server/main.lua (hunk 11 above)
Comment (solareon) [reply server/main.lua]: Changed. Still getting the hang of Lua best practices.

File: server/main.lua (hunk 12 above)
Comment (solareon) [reply server/main.lua]: Changed

File: server/main.lua (hunk 13 above)
Comment (solareon) [reply server/main.lua]: Changed.

Comment (solareon) [reply server/main.lua]: Done and dusted.

File: server/main.lua (hunk 14 above)
Comment (solareon) [reply server/main.lua]: Also done. Fixed as few others where the camelcase rule wasn't applied

Comment (solareon) [reply server/main.lua]: Is there a preference for net events to be lowercase or camelCase?

Comment (BerkieBb) [reply server/main.lua]: camelCase

File: server/main.lua (hunk 15 above)
Comment (solareon) [reply server/main.lua]: What about the cases where we don't use source. is it okay to discard it with _ ?

File: server/main.lua (hunk 16 above)
Comment (solareon) [reply server/main.lua]: done.

File: server/main.lua (hunk 17 above)
Comment (solareon) [reply server/main.lua]: done

File: server/main.lua (hunk 18 above)
Comment (solareon) [reply server/main.lua]: Done

File: client/cl_main.lua (hunk 19 above)
Comment (solareon) [reply client/cl_main.lua]: Done.

**Comment by solareon** [comment]
> I'll add that in the future, please keep your PRs small. This PR is difficult to review now because of it's size.

My apologies for the large PR. Would it help if I squashed the commits down? I didn't intend to rewrite the whole thing but kind of got in the groove after the first couple changes.

I made all the changes regarding the camelCase, localizing functions where appropriate, changing type to group, and only localizing source where appropriate.

File: server/main.lua (hunk 15 above)
Comment (Manason) [reply server/main.lua]: Yes. It is preferred to discard it with "_" when not using it.

File: server/main.lua [hunk 20]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
 	end)
 
 	return players
-end
+end)
+
+---@param accountName string
+---@param group 'job'|'gang'
+---@return table?
+lib.callback.register('qbx_management:server:getemployees', function(source, accountName, group)
```
Comment (Manason) [inline server/main.lua:185]:
```suggestion
lib.callback.register('qbx_management:server:getEmployees', function(source, accountName, group)
```

File: server/main.lua [hunk 21]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
+	return getEmployees(source, accountName, group)
+end)
+
+---@param data table
+---@return table?
+RegisterNetEvent('qbx_management:server:updateGrade', function(data)
```
Comment (Manason) [inline server/main.lua:191]: Should this be a callback returning nil instead? Events should describe something that has happened in the past https://manason.github.io/effective-fivem-lua/events/#past-tense

File: server/main.lua [hunk 22]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
+	updateGrade(source, data)
+	TriggerClientEvent('qbx_management:client:openMenu', source, data.group)
+end)
 
+---@param data {source: number, grade: number, group: 'job'|'gang'}
+RegisterNetEvent('qbx_management:server:HireEmployee', function(data)
```
Comment (Manason) [inline server/main.lua:197]:
```suggestion
RegisterNetEvent('qbx_management:server:hireEmployee', function(data)
```

Comment (Manason) [inline server/main.lua:197]: Ditto on callback

File: server/main.lua [hunk 23]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
-        exports.qbx_core:Notify(src, "Grade does not exist.", "error")
+        exports.qbx_core:Notify(src, 'Grade does not exist.', 'error')
     end
 end
 
----@param src number
+---@param closePlayers table List of nearby server ids
 ---@return table
-function GetPlayers(src)
+lib.callback.register('qbx_management:server:getplayers', function(_, closePlayers)
```
Comment (Manason) [inline server/main.lua:161]:
```suggestion
lib.callback.register('qbx_management:server:getPlayers', function(_, closePlayers)
```

File: server/main.lua [hunk 24]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
+
+    local success = data.group == 'gang' and target.Functions.SetGang(jobName, data.grade) or target.Functions.SetJob(jobName, data.grade)
+    local grade = data.group == 'gang' and gangs[jobName].grades[data.grade].name or jobs[jobName].grades[data.grade].name
+
+    if success then
+        local playerFullName = player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname
+        local targetFullName = target.PlayerData.charinfo.firstname .. ' ' .. target.PlayerData.charinfo.lastname
+        local organizationLabel = player.PlayerData[data.group].label
+
+        exports.qbx_core:Notify(source, 'You hired ' .. targetFullName .. ' into ' .. organizationLabel, 'success')
```
Comment (Manason) [inline server/main.lua:222]: Optional; Consider using string.format to make this a bit easier to read
```suggestion
        exports.qbx_core:Notify(source, 'You hired %s into %s':format(targetFullName, organizationLabel), 'success')
```

File: server/main.lua [hunk 25]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
+			return
+		end
+	
+		local success = data.group == 'gang' and employee.Functions.SetGang('none', 0) or employee.Functions.SetJob('unemployed', 0)
+		if not success then
+			exports.qbx_core:Notify(src, 'Unable to fire citizen.', 'error')
+			return
+		end
+	
+		local notifyMessage = data.group == 'gang' and 'You have been expelled from the gang!' or 'You have been fired! Good luck.'
```
Comment (Manason) [inline server/main.lua:262]: Was this previously not localized?

File: server/main.lua [hunk 26]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
+        exports.qbx_core:Notify(target.PlayerData.source, 'You have been hired into ' .. organizationLabel, 'success')
+        TriggerEvent('qb-log:server:CreateLog', logArea..'menu', grade, 'yellow', playerFullName.. ' successfully recruited ' .. targetFullName .. ' (' .. organizationLabel .. ')', false)
+    else
+        exports.qbx_core:Notify(source, 'Couldn\'t hire civilian', 'error')
+    end
+	TriggerClientEvent('qbx_management:client:openMenu', source, data.group)
+end)
+
+---@param data {source: number, group: 'job'|'gang'}
+RegisterNetEvent('qbx_management:server:fireEmployee', function(data)
```
Comment (Manason) [inline server/main.lua:232]: This is a large function. Consider if it can be broken up into smaller functions.

File: server/main.lua [hunk 27]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
+		MySQL.update(string.format('UPDATE players SET %s = ? WHERE citizenid = ?', updateColumn), {json.encode(role), data.source})
+		exports.qbx_core:Notify(src, logType, 'success')
+		TriggerEvent('qb-log:server:CreateLog', logArea..'menu', data.group..' Fire', 'orange', playerFullName .. ' successfully fired ' .. employeeFullName .. ' (' .. organizationLabel .. ')', false)
+	end
+	TriggerClientEvent('qbx_management:client:openMenu', src, data.group)
+end)
+
+---@param group 'job'|'gang'
+---@param amount number
+RegisterNetEvent('qbx_management:server:withdrawMoney', function(group, amount)
```
Comment (Manason) [inline server/main.lua:305]: Seems like this should be a callback

File: server/main.lua [hunk 28]
```diff
@@ -154,75 +128,214 @@ function GetEmployees(src, accountName, type)
...
+	if not success or not player then return end
+	local logArea = group == 'gang' and 'gang' or 'boss'
+	
+	TriggerEvent('qb-log:server:CreateLog', logArea..'menu', 'Withdraw Money', 'yellow', player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname .. ' successfully withdrew $' .. amount .. ' (' .. account .. ')', false)
+	TriggerClientEvent('qbx_management:client:openMenu', source, group)
+end)
+
+---@param group 'job'|'gang'
+---@param amount number
+RegisterNetEvent('qbx_management:server:depositMoney', function(group, amount)
```
Comment (Manason) [inline server/main.lua:316]: Also seems like it should be a callback

File: client/cl_main.lua [hunk 29]
```diff
@@ -0,0 +1,316 @@
+local jobs = exports.qbx_core:GetJobs()
+local gangs = exports.qbx_core:GetGangs()
+
+local function commaValue(amount)
```
Comment (Manason) [inline client/cl_main.lua:4]: I think core utils has a comma value util?

File: client/cl_main.lua [hunk 30]
```diff
@@ -0,0 +1,316 @@
...
+end)
+
+---@param group 'job'|'gang'
+RegisterNetEvent('qbx_management:client:hireMenu', function(group)
+    local hireMenu = {}
+    local players = findPlayers()
+    local hireName = QBX.PlayerData[group].name
+    for _, v in pairs(players) do
+
+        if v and v.citizenid ~= QBX.PlayerData.citizenid and v[group].name ~= hireName then
```
Comment (Manason) [inline client/cl_main.lua:284]: players will no longer include the client, so no need to check that citizen is not equals. Additionally, it is impossible for the value of a table loops through using pairs to be nil.
```suggestion
        if v[group].name ~= hireName then
```

File: client/cl_main.lua [hunk 31]
```diff
@@ -0,0 +1,316 @@
...
+
+    lib.showContext('qbx_management_open_Member')
+end)
+
+---@param group 'job'|'gang'
+RegisterNetEvent('qbx_management:client:hireMenu', function(group)
+    local hireMenu = {}
+    local players = findPlayers()
+    local hireName = QBX.PlayerData[group].name
+    for _, v in pairs(players) do
```
Comment (Manason) [inline client/cl_main.lua:282]: Consider a more descriptive variable name than 'v'
```suggestion
    for _, player in pairs(players) do
```

**Review by Manason — CHANGES_REQUESTED** [review]

**Comment by Manason** [comment]
> > I'll add that in the future, please keep your PRs small. This PR is difficult to review now because of it's size.
> 
> My apologies for the large PR. Would it help if I squashed the commits down? I didn't intend to rewrite the whole thing but kind of got in the groove after the first couple changes.
> 
> I made all the changes regarding the camelCase, localizing functions where appropriate, changing type to group, and only localizing source where appropriate.

That would make it worse actually. In the future please keep your PRs focused on one change. You can branch off your branch to create a second PR, and branch off of that branch to create a third, etc..

File: server/main.lua (hunk 25 above)
Comment (solareon) [reply server/main.lua:262]: No. There was no locales in this resource.

Comment (solareon) [reply server/main.lua:262]: No there wasn't any translations in this resource.

**Comment by solareon** [comment]
Closing this PR to submit a new one due to full rewrite.
